In [2]:
from pathlib import Path
import sys

project_root = Path.cwd().parent

if str(project_root) not in sys.path:
    sys.path.append(
        str(project_root)
    )

In [3]:
import pandas as pd
import matplotlib.pyplot as plt

from src.data_ingestion import (
    build_marketplace_base
)

# Business Recommendations | Recomendações de Negócio

### 🇧🇷 Português

Este notebook converte análises anteriores em decisões estratégicas de negócio.

Objetivos:

- Medir risco comercial;
- Avaliar concentração de sellers;
- Identificar oportunidades de crescimento.

---

### 🇺🇸 English

This notebook converts previous analysis into strategic business decisions.

Objectives:

- Measure commercial risk;
- Evaluate seller concentration;
- Identify growth opportunities.

In [4]:
df = build_marketplace_base()

df.shape

Marketplace base created successfully.
Rows: 112650
Columns: 16


(112650, 16)

In [5]:
seller_revenue = (
    df.groupby("seller_id")["payment_value"]
    .sum()
    .sort_values(ascending=False)
)

seller_revenue_df = (
    seller_revenue
    .reset_index()
)

seller_revenue_df.columns = [
    "seller_id",
    "revenue"
]

In [10]:
seller_revenue_df["segment"] = pd.qcut(
    seller_revenue_df["revenue"],
    q=4,
    labels=[
        "Long Tail",
        "Emerging",
        "Strong",
        "Top Performers"
    ]
)

seller_revenue_df.head()

,seller_id,revenue,segment
0,7c67e1448b00f6e969d365cea6b010ab,507166.91,Top Performers
1,1025f0e2d44d7041d6cf58b6550e0bfa,308222.04,Top Performers
2,4a3ca9315b744ce9f8e9374361493884,301245.27,Top Performers
3,1f50f920176fa81dab994f9023523100,290253.42,Top Performers
4,53243585a1d6dc2643021fd1853d8905,284903.08,Top Performers


In [11]:
seller_revenue_df.columns

Index(['seller_id', 'revenue', 'segment'], dtype='str')

# Business Recommendations | Recomendações de Negócio

## Revenue Risk Matrix | Matriz de Risco de Receita

### 🇧🇷 Português

Após identificar concentração de receita entre sellers, o próximo passo é medir o risco comercial do marketplace.

Pergunta central:

Quanto da receita está concentrada nos sellers de alta performance?

Objetivo:

Avaliar dependência comercial e exposição a churn estratégico.

---

### 🇺🇸 English

After identifying revenue concentration across sellers, the next step is to measure marketplace commercial risk.

Core question:

How much revenue is concentrated among top-performing sellers?

Objective:

Evaluate commercial dependency and exposure to strategic churn.

In [13]:
segment_revenue_pct = (
    seller_revenue_df
    .groupby("segment")["revenue"]
    .sum()
)

segment_revenue_pct = (
    segment_revenue_pct
    / segment_revenue_pct.sum()
) * 100

segment_revenue_pct = (
    segment_revenue_pct
    .round(2)
)

segment_revenue_pct

segment
Long Tail          0.57
Emerging           2.70
Strong             9.93
Top Performers    86.79
Name: revenue, dtype: float64

## Revenue Risk Matrix | Matriz de Risco de Receita

### 🇧🇷 Português

A análise revelou forte concentração comercial entre sellers.

Distribuição da receita:

- **Long Tail:** 0.57%
- **Emerging:** 2.70%
- **Strong:** 9.93%
- **Top Performers:** 86.79%

### Insight estratégico:

O marketplace apresenta alta dependência estrutural dos sellers Top Performers.

### Riscos identificados:

- Churn de sellers estratégicos;
- Perda de market share;
- Vulnerabilidade de receita.

### Recomendação:

Criar programas de:

- retenção;
- account management;
- relacionamento estratégico com sellers premium.

---

### 🇺🇸 English

The analysis revealed strong commercial concentration across sellers.

Revenue distribution:

- **Long Tail:** 0.57%
- **Emerging:** 2.70%
- **Strong:** 9.93%
- **Top Performers:** 86.79%

### Strategic insight:

The marketplace shows strong structural dependency on Top Performers.

### Identified risks:

- Strategic seller churn;
- Market share loss;
- Revenue vulnerability.

### Recommendation:

Build programs for:

- retention;
- account management;
- strategic relationship management with premium sellers.